### PR TITLE
[NOP] minor optimization to FeatureFinderMultiplex

### PR DESCRIPTION
**Pull request by timosachsenberg**
Resulted in major speedup (factor ~4) for a dataset I tested.

### CODE REVIEW

File: src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h [hunk 1]
```diff
@@ -539,17 +539,23 @@ namespace OpenMS
     {
       int A1 = c1.getPropertyA();
       int A2 = c2.getPropertyA();
-      std::vector<int> B1 = c1.getPropertiesB();
-      std::vector<int> B2 = c2.getPropertiesB();
 
       // check if any of the properties A and B is not set i.e. =-1
```
Comment (lars20070) [inline src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h]: Please update comment.

// Check if the properties A of both clusters have been set or not. (i.e. =-1 if not set)

File: src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h [hunk 2]
```diff
@@ -539,17 +539,23 @@ namespace OpenMS
...
         return false;
       }
 
       // Will the merged cluster have the same properties A?
-      bool vetoA = !(A1 == A2);
+      if (A1 != A2) return true;
+
+      std::vector<int> B1 = c1.getPropertiesB();
+      std::vector<int> B2 = c2.getPropertiesB();
+
```
Comment (lars20070) [inline src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h]: Please add comment.

// Check if the properties B of both clusters have been set or not. (i.e. =-1 if not set)

**Comment by lars20070** [comment]
+1

File: src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h [hunk 3]
```diff
@@ -577,8 +582,8 @@ namespace OpenMS
      */
     bool findNearestNeighbour_(GridBasedCluster cluster, int cluster_index)
```
Comment (cbielow) [inline src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h]: const ref for GridBasedCluster?! Its quite a large beast...

**Comment by cbielow** [comment]
also GridBasedCluster.getPoints() should return const& 

when .getPoints() is used within GridBasedClustering, there is also potential for optimization (lots of copying going on, just for a simple merge).

File: src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h (hunk 3 above)
Comment (lars20070) [inline src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h]: +1

**Comment by hroest** [comment]
:+1: looks great

**Pull request merged**